pkg/utils/utils.go

```diff
@@ -110,6 +110,12 @@ func ErrorContains(out error, want string) bool {
 // We MUST NOT check the name or key property here. It MAY be defaulted by the provider.
 func ValidateSecretSelector(store esv1beta1.GenericStore, ref esmeta.SecretKeySelector) error {
 	clusterScope := store.GetObjectKind().GroupVersionKind().Kind == esv1beta1.ClusterSecretStoreKind
+	if ref.Name == "" {
```

Comment:
see above:
```
// We MUST NOT check the name or key property here.
// It MAY be defaulted by the provider.
```

AFAIK provider-level defaulting is only implemented by the vault provider

Reply:
i guess we can set the ref.Name on the provider and then check the `Name`/`Key`.

Reply:
i guess we can set the ref.Name on the provider and then check the `Name`/`Key`.

Reply:
I believe changing Vault Provider behavior might be tricky. The secretRef is manipulated throughout several pieces of the code.

It's simpler to just put this validation in IBM provider.